server/src/main/java/io/crate/execution/dml/Indexer.java

```diff
@@ -117,6 +120,12 @@ public class Indexer {
     private final Function<ColumnIdent, Reference> getRef;
     private final boolean writeOids;
     private final Version tableVersionCreated;
+    @Nullable
+    private final UpdateToInsert updateToInsert;
+    private final List<Reference> assignedColumns;
```

Comment:
I think it would be worth adding some javadocs that explain assignedColumns/columns and indexingOrder.
Mostly to get a sense of what the difference between them is.